rialto_airflow/publish/openaccess.py

```diff
@@ -163,6 +163,154 @@ def write_contributions(snapshot) -> Path:
     return csv_path
 
 
+def write_contributions_by_school(snapshot) -> Path:
+    """
+    Write a CSV of contributions where each row represents a unique publication per school.
+    """
+
+    col_names = [
+        "academic_council_authored",
+        "apc",
+        "doi",
+        "faculty_authored",
+        "federally_funded",
+        "open_access",
+        "primary_school",
+        "pub_year",
+        "types",
+    ]
+
+    csv_path = snapshot.path / "contributions-by-school.csv"
+
+    logging.info(f"starting to write contributions by school {csv_path}")
+
+    with csv_path.open("w") as output:
+        csv_output = DictWriter(output, fieldnames=col_names)
+        csv_output.writeheader()
+
+        with get_session(snapshot.database_name).begin() as session:
+            stmt = (
+                select(  # type: ignore
+                    Publication.apc,  # type: ignore
+                    Publication.doi,  # type: ignore
+                    Publication.open_access,  # type: ignore
+                    Author.primary_school,
+                    Publication.pub_year,  # type: ignore
+                    # for academic_council
+                    func.jsonb_agg_strict(Author.academic_council).label(
+                        "academic_council"
+                    ),
+                    # for publication types
+                    Publication.dim_json["type"].label("dim_type"),
+                    Publication.openalex_json["type"].label("openalex_type"),
+                    # for federally_funded
+                    func.jsonb_agg_strict(Funder.federal).label("federal"),
+                    # for faculty_authored
+                    func.jsonb_agg_strict(Author.primary_role).label("roles"),
+                )
+                .join(Author, Publication.authors)  # type: ignore
+                .join(Funder, Publication.funders, isouter=True)  # type: ignore
```

Comment:
just curious what the `type: ignore` comment is (seen it in other places but not sure what it means)